pkg/controller/registry/reconciler/grpc.go

```diff
@@ -186,17 +186,34 @@ func (c *GrpcRegistryReconciler) currentPodsWithCorrectImage(source grpcCatalogS
 	return found
 }
 
+func (c *GrpcRegistryReconciler) currentPodsWithCorrectHash(source grpcCatalogSourceDecorator, saName string) []*corev1.Pod {
+	pods, err := c.Lister.CoreV1().PodLister().Pods(source.GetNamespace()).List(labels.SelectorFromValidatedSet(source.Labels()))
```

Comment:
Is it worth the effort on refactoring this and the above method to avoid making the same pod listing call, and instead attempt to return a filtered set of pods given a pod list parameter being provided at the call site? You would then make the list command at the EnsureRegistryServer call site, pass that podList variable to both of the currentPodWith* methods. This can likely be tackled in a follow-up though as I think you'd want a more functional refactoring than what I had suggested.

Reply:
I was thinking the same as well but I decided not to combine due to the fact that func is for something else that is not necessarily relevant to this fix. However, on a second thought, I will give it a shot.